chain/chain/src/chain.rs

```diff
@@ -2136,21 +2136,29 @@ impl Chain {
             return Ok(None);
         }
 
+        let shard_layout = self.epoch_manager.get_shard_layout(epoch_id)?;
+
         let last_final_block = self.get_block(&last_final_block_hash)?;
         let last_final_block_epoch_id = last_final_block.header().epoch_id();
         // If shard layout was changed, the update is impossible so we skip
         // getting candidate.
-        if self.epoch_manager.get_shard_layout(last_final_block_epoch_id)
-            != self.epoch_manager.get_shard_layout(epoch_id)
-        {
+        if self.epoch_manager.get_shard_layout(last_final_block_epoch_id)? != shard_layout {
             return Ok(None);
         }
 
+        let shard_index = shard_layout.get_shard_index(shard_uid.shard_id())?;
```

Comment:
Please add short comment why we need it. I'd be tempted to use only shard id check but don't want to rely on shard id uniqueness.

Reply:
Yea I was thinking of just comparing the ShardIds since we're going with unique ones anyway, but I got paranoid and went with the full shard UID check